axlearn/common/module.py

```diff
@@ -535,6 +535,12 @@ def _wrap_method_with_auto_child_context(*, method_fn: Callable, method_name: st
     `partial(method_fn, instance)`, or supply an instance explicitly as the first arg.
     """
 
+    if not traceback_util.is_stack_summary_enabled():
+        method_fn = functools.wraps(method_fn)(
+            functools.partial(_call_method_in_context, method_fn=method_fn, method_name=method_name)
+        )
```

Comment:
Nit: since we always `_call_method_in_context`, maybe we can move it outside the `if`:

```
method_fn_in_context = functools.partial(
    _call_method_in_context, method_fn=method_fn, method_name=method_name)
if not traceback_util.is_stack_summary_enabled():
    return functools.wraps(method_fn)(method_fn_in_context)
```